packages/dev/loaders/src/glTF/glTFFileLoader.ts

```diff
@@ -562,12 +562,26 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
      */
     public loadFile(
         scene: Scene,
-        fileOrUrl: File | string,
+        fileOrUrl: File | string | ArrayBufferView,
         onSuccess: (data: any, responseURL?: string) => void,
         onProgress?: (ev: ISceneLoaderProgressEvent) => void,
         useArrayBuffer?: boolean,
-        onError?: (request?: WebRequest, exception?: LoadFileError) => void
+        onError?: (request?: WebRequest, exception?: LoadFileError) => void,
+        name?: string
     ): IFileRequest {
+
+        if(fileOrUrl as ArrayBufferView)
+        {
+            this._loadBinary(scene, fileOrUrl as ArrayBufferView, onSuccess, onError, name);
+            
+            const fileRequest: IFileRequest = {
+                abort: () => {},
+                onCompleteObservable: new Observable<IFileRequest>(),
+            };
+            
+            return fileRequest;

```

Comment:
is there any use for this fileRequest object, or is it just here because you need to return a IFileRequest? the observable won't be notified.

Reply:
It's the contract of the loadFile function in the plugin interface. We could maybe make this return null or something to indicate it's not a file.

Reply:
It's technically the same with the abort function as well. Not sure what would be the best approach here, but there is no real need to return anything if it is sync-loaded (like in this case). Abort won't do anything post-call, and the observable will only be created and destructed

Reply:
I've updated loadFile to return a Nullable<IFileRequest>, that's already how sceneLoader is managing it. If the request is null it will skip request.abort().